docs: guide for visualization with Isaac Sim using omnigraph (#429)
aica-technology/api: docs/core/examples/guides/omnigraph-aica-bridge.md

multi-sensor support, and RTX-based rendering, making it ideal for developing, testing, and validating robot software
before deploying to real hardware.

To connect Isaac Sim with AICA Core, we will use **ROS 2 Bridge**, an extension that lets Isaac Sim publish and
subscribe to ROS 2 topics and services. Through this bridge, we can exchange robot state, control commands, sensor data,
and more between the simulator and AICA Core.

**OmniGraph** is a visual, node-based programming system integrated into Isaac Sim. It lets you assemble logic and data
flows, called _Action Graphs_, by connecting pre-built nodes. OmniGraph can include ROS 2 Bridge nodes, enabling
communication between Isaac Sim and ROS 2 without writing code.

This interface can be used in two main ways:

1. **Control robots in Isaac Sim from AICA Core:** In this mode, AICA Core sends motion commands and control inputs to
   the simulated robot via ROS 2, and Isaac Sim returns robot state and sensor feedback. From AICA’s point of view, the
   robot in Isaac Sim behaves just like real hardware, which is ideal for validating control algorithms before deploying
   them to physical robots.

> 2. **Visualize robots in Isaac Sim from AICA Core:** Here, the robot is controlled by AICA Core (e.g., in a production

Reviewer: I think this lacks precision. In both modes, the *robot* is controlled by "an AICA application" or "an application running in AICA Studio". The difference is that 1) the robot is once virtual and once real and 2) that we feed back the state or not

   environment), and Isaac Sim mirrors the robot’s real-world pose and joint states for visualization. This setup
   provides a live, virtual view of what the robot is doing on the factory floor, helping with monitoring, debugging,
   and demonstration.

## Prerequisites

Begin by installing Isaac Sim using the official
[installation instructions](https://docs.isaacsim.omniverse.nvidia.com/latest/installation/index.html). AICA Core runs
on **ROS 2 Jazzy**, so be sure to install a version of Isaac Sim that supports this distribution (**Isaac Sim v5.0 or
later**).

Isaac Sim supports multiple installation options, including workstation installation, container-based deployment, and
cloud deployment. Choose the method that best fits your workflow. For this guide, we assume you have installed Isaac Sim
locally using the

> **[workstation installation](https://docs.isaacsim.omniverse.nvidia.com/5.1.0/installation/install_workstation.html)**

Reviewer: Link and bold seems too much emphasize IMO

  the robot state and sensor data (e.g., joint states, poses, perception). From the AICA application’s point of view, the
  simulated robot behaves like real hardware, making this setup well suited for validating and debugging control algorithms 
  before deploying them to a physical robot.

2. **Visualize a robot in Isaac Sim from an AICA application:** In this mode, an application running in AICA Studio controls
  a robot that is not in Isaac Sim (e.g., in a production environment, AICA's mock interface, URSim, ...). The robot’s state 
  (such as joint positions) is streamed to Isaac Sim, which mirrors the robot’s motion in a virtual scene. Isaac Sim 
  is used purely for visualization and does not participate in the control loop. This provides a live digital view of 
  the robot for monitoring, debugging, and demonstration purposes.

## Prerequisites

Begin by installing Isaac Sim using the official
[installation instructions](https://docs.isaacsim.omniverse.nvidia.com/latest/installation/index.html). For AICA Core v5,
ensure you have Isaac Sim v5 or later.

> Isaac Sim supports multiple installation options, including workstation installation, container-based deployment, and

Reviewer 2: personal preference:

```suggestion
Isaac Sim supports multiple installation options, including workstation installation, container-based and

cloud deployment. Choose the method that best fits your workflow. For this guide, we assume you have installed Isaac Sim
locally using the
[workstation installation](https://docs.isaacsim.omniverse.nvidia.com/5.1.0/installation/install_workstation.html)
method.

After installation, launch Isaac Sim with the ROS 2 bridge enabled and set **ROS 2 Jazzy** as the active ROS
distribution. To do this, run the following command:

```bash
cd <path-to-isaac-sim>
./isaac-sim.selector.sh
```

<div class="text--center">
  <img src={selector} alt="NVIDIA Isaac Sim App Selector" />
</div>

This command opens the Isaac Sim launcher. In the launcher:

- Set ROS Bridge Extension to `isaacsim.ros2.bridge`

> - Set Use Internal ROS2 Libraries to `jazzy`

Reviewer 2: Not a strong preference. Initially I didn't see the image this is why I commented in the first place.

            x: 1060
            y: 20
      buttons:
        button:
          display_name: Trigger Events Button
          on_click:
            call_service:
              controller: joint_trajectory_controller
              hardware: hardware
              service: set_trajectory
              payload: '{"times_from_start": [1, 2, 5], "frames": ["wp1", "wp2", "wp3"]}'
      edges:
        on_start_on_start_joint_signal_to_joint_state_message_joint_signal_to_joint_state_message:
          path:
            - x: 280
              y: 60
            - x: 280
              y: 200
        on_start_on_start_hardware_hardware:
          path:
            - x: 540
              y: 60
            - x: 540
              y: 80
        hardware_hardware_robot_state_broadcaster_joint_state_joint_signal_to_joint_state_message_input:
          path:
            - x: 440
              y: 480
            - x: 440
              y: 360
        hardware_hardware_joint_trajectory_controller_has_trajectory_succeeded_hardware_hardware_joint_trajectory_controller_set_trajectory:
          path:
            - x: 960
              y: 780
            - x: 960
              y: 860
    ```
</details>

> Copy the above YAML content into a `New Application` in AICA StudIo and save it. This application will use the Generic

Reviewer 2: If this is meant to put emphasis on New Application, why not use bold instead?

`Topic Name` field to `/joint_command` to match the topic used by AICA Studio for the Generic robot.

-3- **Articulation Controller**: This node is responsible for controlling the robot's joints based on the received
joint states. Select the Generic robot in the scene as the `Articulation` for this node.

-4- **On Playback Tick**: This node triggers the graph execution on each simulation tick.

Now that you have all the necessary nodes, you can connect them as follows:

- Connect the `Context` signal on `ROS 2 Context` node's output to the `Context` signal on the
  `ROS2 Subscribe Joint State` node's input.
- Connect the `Joint Names` output of the `ROS2 Subscribe Joint State` node to the `Joint Names` input of the
  `Articulation Controller` node.
- Connect the `Joint Positions` output of the `ROS2 Subscribe Joint State` node to the `Position Command` input of the
  `Articulation Controller` node.
- Connect the `Tick` output of the `On Playback Tick` node to the `Exec` input of the `ROS2 Subscribe Joint State` node
  and the `Exec` input of the `Articulation Controller` node.

Your OmniGraph should look similar to the image below:

> <div class="text--center">

Author: let me try to expand a bit more the screen when taking the screenshot

                  hardware: hardware
                - controller: joint_trajectory_controller
                  hardware: hardware
        controllers:
          robot_state_broadcaster:
            plugin: aica_core_controllers/RobotStateBroadcaster
            outputs:
              joint_state: /hardware/robot_state_broadcaster/joint_state
            events:
              transitions:
                on_load:
                  switch_controllers:
                    hardware: hardware
                    activate: robot_state_broadcaster
            display_name: Robot State Broadcaster
          joint_trajectory_controller:
            plugin: aica_core_controllers/trajectory/JointTrajectoryController
            events:
              transitions:
                on_load:
                  switch_controllers:
                    hardware: hardware
                    activate: joint_trajectory_controller
              predicates:
                has_trajectory_succeeded:
                  call_service:
                    controller: joint_trajectory_controller
                    hardware: hardware
                    service: set_trajectory

>                     payload: '{"times_from_start": [1, 2, 5], "frames": ["wp1", "wp2", "wp3"]}'

Reviewer: ```suggestion
                    payload: '{"times_from_start": [5, 2, 1], "frames": ["wp1", "wp2", "wp3"]}'
```
You're using 1 second for the biggest displacement, I find that this just looks a bit unrealistic

          path:
            - x: 440
              y: 480
            - x: 440
              y: 360
        hardware_hardware_joint_trajectory_controller_has_trajectory_succeeded_hardware_hardware_joint_trajectory_controller_set_trajectory:
          path:
            - x: 960
              y: 780
            - x: 960
              y: 860
    ```
</details>

Copy the above YAML content into a **New Application** in AICA Studio and save it. This application will use the `Generic`
robot model and send joint commands to Isaac Sim via the `/joint_state` topic.

This component is key the bridge between AICA Studio and Isaac Sim as it converts the joint signals from AICA into ROS 2
JointState messages that Isaac Sim can consume:

```yaml
components:
  joint_signal_to_joint_state_message:
    component: aica_core_components::ros::JointSignalToJointStateMsg
    display_name: Joint Signal To Joint State Message
    inputs:
      input: /hardware/robot_state_broadcaster/joint_state
    outputs:
      output: /joint_state
```

Validate the application by pressing Play in AICA Studio. You will see the robot moving between 3 waypoints defined in

> the application `wp1`, `wp2`, and `wp3`.

Reviewer: ```suggestion
Additionally, the application has a `Joint Signal To Joint State Message` component that converts the robot joint state
to the message type expected by the Isaac Sim ROS bridge and publishes it to the `/joint_state` topic.

\```yaml
components:
  joint_signal_to_joint_state_message:
    component: aica_core_components::ros::JointSignalToJointStateMsg
    display_name: Joint Signal To Joint State Message
    inputs:
      input: /hardware/robot_state_broadcaster/joint_state
    #highlight-next-line
    outputs:
      #highlight-next-line
      output: /joint_state
\```

:::info

Since Isaac Sim doesn't appear as a component in AICA Studio, the output topic of the translator component has to be
edited manually in the code editor. Simply add the two highlighted lines above to configure the output to the matching
topic name.

:::

Validate the application by pressing Play in AICA Studio. The robot moves between 3 waypoints defined in the application
as `wp1`, `wp2`, and `wp3`.
```
Watch out this suggestion has two backslashes around the code block that are not correct.

<div style={{ display: "flex", justifyContent: "center" }}>
  <video autoPlay loop muted playsInline style={{ maxWidth: "100%", borderRadius: "8px" }}>
    <source src={runningAICA} type="video/webm" />
  </video>
</div>
<br/>

Additionally as can be seen in the AICA Studio, the joint commands are retrieved from the current robot state and that's
exactly why this approach is for visualization purposes only. The state of the robot in Isaac Sim is not fed back to
AICA Studio.

## Interfacing Isaac Sim with AICA Studio

Now that both Isaac Sim and AICA Studio are set up, you can run the simulation to see the communication in action.

1. **Start AICA Application**: First, ensure that your AICA application is running. You can do this by pressing the

>    `Start` button in AICA Studio. This will start publishing joint commands to the `/joint_state` topic.

Reviewer: ```suggestion
   `Start` button in AICA Studio. This will start broadcasting the robot's state to the `/joint_state` topic.
```

2. **Start Isaac Sim**: If all the steps above have been followed correctly, you just need to press the `Play` button in
   Isaac Sim to start the simulation. The OmniGraph will start executing, and the robot in the scene will begin

>    receiving joint commands from AICA Studio.

Reviewer: ```suggestion
   Isaac Sim to start the simulation. The OmniGraph will start executing, and the robot in the scene will mirror the one
   from AICA Studio.
```

<br/>

In the OmniGraph editor, you can create nodes and connect them. The following nodes are required to set up the
communication between Isaac Sim and AICA Studio:

-1- **ROS2 Context**: This node initializes the ROS 2 context and allows and defines the ROS2 domain ID. In order to set
the domain ID, double click on the node to open its properties and set the `Domain ID` field to `30`. This domain ID
must match the one used by AICA Studio to ensure proper communication.

-2- **ROS2 Subscribe Joint State**: This node subscribes to the joint state topic published by AICA STudio. Set the
`Topic Name` field to `/joint_command` to match the topic used by AICA Studio for the Generic robot.

> -3- **Arcticulation Controller**: This node is responsible for controlling the robot's joints based on the received

Reviewer 2: just a guess:

```suggestion
-3- **Articulation Controller**: This node is responsible for controlling the robot's joints based on the received
